Jetpack Connect Plans

roccotripaldi opened the pull request:
Updating to newly designed plans layout at `client/my-sites/plans-features-main`

See #7130 

Here's a screencast of the newly designed plan boxes in our Jetpack Connect Flow:
https://cloudup.com/cEatUbI_oHb

And a screenshot:
![jpc-new-plans-page](https://cloud.githubusercontent.com/assets/2694219/17254829/6ef680a8-5584-11e6-826a-61fa25ea450f.jpg)


Currently working on making the Free plan selectable. And it will need some design.
cc: @lamosty 

Test live: https://calypso.live/?branch=update/jetpack-connect-plans

roccotripaldi commented:
cc: @rickybanister for design review

lamosty commented:
cc @gwwar for more help. :)

gwwar commented:
From the screens it looks like we'll need to update the constants file to add details for the free jetpack plan in https://github.com/Automattic/wp-calypso/blob/master/client/lib/plans/constants.js#L153 This was left empty since we never rendered this on any other page.

cc @bubel for free features/copy ideas

gwwar commented on client/my-sites/plans-features-main/index.jsx:
```diff
@@ -41,17 +42,25 @@ class PlansFeaturesMain extends Component {
 		const isPersonalPlanEnabled = isEnabled( 'plans/personal-plan' );
 
 		if ( this.isJetpackSite( site ) && intervalType === 'monthly' ) {
+			let jetpackPlans = [ PLAN_JETPACK_FREE, PLAN_JETPACK_PREMIUM_MONTHLY, PLAN_JETPACK_BUSINESS_MONTHLY ];
```
This can be a const since we don't reassign the variable.

gwwar commented:
Note that we have an outstanding bug for the mobile view https://github.com/Automattic/wp-calypso/issues/7170

gwwar commented:
@roccotripaldi to make the free plan selectable update https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/plan-features/index.jsx#L328, https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/plan-features/index.jsx#L244, and https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/plan-features/index.jsx#L129 to also check for the Jetpack free plan

gwwar commented:
When #7173 lands this should fix #7170 and the free plan selection issue.

gwwar commented:
Fixes are in, so rebase with master when you're ready

rickybanister commented:
This looks great to me, except for the Free plan. 

I'm not sure it makes sense to say 'Free $0 for life' 

Previously we never showed a dollar amount (if I recall correctly) next to the free plan. 

Is anyone else bothered by that? If so I can think about proposing an alternative layout for just that plan.

bubel commented:
I'm not really bothered by the current format, @rickybanister, but if you'd like to propose something else, go for it.

lamosty commented:
I'm having difficulty in testing this. I've disconnected my wpcom Jetpack testing site and now can't reconnect. After going to http://calypso.localhost:3000/jetpack/connect and typing in my site, I'm redirected to http://calypso.localhost:3000/jetpack/connect/authorize with the following error:

![selection_097](https://cloud.githubusercontent.com/assets/4988512/17344116/481eafca-5901-11e6-9a79-9d05c26993fd.png)

When I reload my Jetpack testing site's Jetpack settings afterwards, it says I'm connected (and it can be verified in WP.com sites).

Question: how do you usually test these things? What am I doing wrong? :)

roccotripaldi commented:
@lamosty - you seem to be connecting properly. maybe there is a glitch. i'll see if i can reproduce

lamosty commented on client/lib/plans/constants.js:
```diff
@@ -437,6 +447,22 @@ export const featuresList = {
+	},
+	[ FEATURE_MANAGE ]: {
+		getTitle: () => i18n.translate( 'Centralized Dashboard' ),
+		getDescription: () => i18n.translate( 'Manage all of your WordPress sites from one location.' )
```
Could you add `getSlug` method to these please? Like this: https://github.com/Automattic/wp-calypso/blob/master/client/lib/plans/constants.js#L441. It's currently used for feature highlighting added in this PR: https://github.com/Automattic/wp-calypso/pull/7188.

lamosty commented on client/my-sites/plans-features-main/index.jsx:
```diff
@@ -42,23 +43,25 @@ class PlansFeaturesMain extends Component {
 		}
 
 		if ( this.isJetpackSite( site ) ) {
+			let jetpackPlans = [ PLAN_JETPACK_FREE, PLAN_JETPACK_PREMIUM, PLAN_JETPACK_BUSINESS ];
```
Can be const too.

lamosty commented on client/signup/jetpack-connect/plans.jsx:
```diff
@@ -156,16 +85,10 @@ const Plans = React.createClass( {
-								onSelectPlan={ this.selectPlan }
-								onSelectFreeJetpackPlan={ this.selectFreeJetpackPlan }/>
+								intervalType="yearly"
+								hideFreePlan={ false } />
```
This is probably redundant as `hideFreePlan` is set to `false` by default here: https://github.com/Automattic/wp-calypso/blob/78bb5f44accad894b8d1d3ec0e3ee11b5d94a5d7/client/my-sites/plans-features-main/index.jsx#L298-L298.

lamosty commented:
Looks and works quite nicely. Great work!

One thing I noticed is that you don't use the wide layout of `Main`. Add `isWideLayout` to `Main` here, please: https://github.com/Automattic/wp-calypso/blob/78bb5f44accad894b8d1d3ec0e3ee11b5d94a5d7/client/signup/jetpack-connect/plans.jsx#L76-L76.

lamosty commented:
> you seem to be connecting properly. maybe there is a glitch. i'll see if i can reproduce

Sorry, I forgot to update Jetpack on my testing site. All good now.

lamosty commented on client/lib/plans/constants.js:
```diff
@@ -437,6 +447,22 @@ export const featuresList = {
 	},
+	[ FEATURE_STANDARD_SECURITY_TOOLS ]: {
+		getTitle: () => i18n.translate( 'Standard Security Tools' ),
+		getDescription: () => i18n.translate( 'Brute force protection, uptime monitoring, secure sign on, and automatic updates for your plugins.' )
```
Can we split this to two lines? It's quite long.

roccotripaldi commented:
@lamosty I addressed your feedback, and I added the ability to select the free plan.

Question for @rickybanister :  should we show the FAQ section in this flow? We do have the option to hide it easily.

cc: @johnHackworth for a code review.  Javi this PR is of note to you as you are attempting the plans first approach in #6843 - In this PR, I've removed `autoselectPlan` because I wasn't sure how to handle it. But I think we'll need to add it back before we can merge this. Thanks for the reviews everyone. :)

johnHackworth commented:
yup, we can't remove `autoselectPlan` because that would mess with the /premium & /pro routes ... I'll take a look

rickybanister commented:
@roccotripaldi maybe we test it with and without FAQ? 10% to each variation?

roccotripaldi commented:
> maybe we test it with and without FAQ? 10% to each variation?

I think we should add the test in a separate PR. To start, we'll keep the FAQ in.
This is ready for review. I'll add some testing instructions in a bit.

lamosty commented on client/signup/jetpack-connect/plans.jsx:
```diff
@@ -60,6 +60,39 @@ const Plans = React.createClass( {
+				user: this.props.userId
+			} );
+		}
+		upgradesActions.addItem( cartItem );
```
Is this necessary? Couldn't we just redirect to `page( `/checkout/${ selectedSiteSlug }/${ getPlanPath( plan ) || '' }` );` as done here: https://github.com/Automattic/wp-calypso/blob/533265f38693c5349a8acd75f323f558fd4fe766/client/my-sites/plan-features/index.jsx#L389-L389?

roccotripaldi replied:
I tried that, but in this context we do not have access to `plan`, so we cannot `getPlanPath( plan )`
Is this a problem? Are we getting rid of `upgradeActions`

lamosty commented on client/my-sites/plan-features/actions.jsx:
```diff
@@ -32,7 +33,7 @@ const PlanFeaturesActions = ( {
 	);
 
-	if ( current ) {
+	if ( current && ! isInSignup ) {
```
I wonder: we need this because a Jetpack site already has a free plan assigned? Because we use the same code in Calypso signup (http://calypso.localhost:3000/start) and it works without having to pass down `isInSignup`.

roccotripaldi replied:
I believe this is the case. Jetpack sites already have the free plan assigned.

lamosty commented on client/signup/jetpack-connect/plans.jsx:
```diff
@@ -73,7 +106,7 @@ const Plans = React.createClass( {
 		return (
 			<div>
-				<Main>
+				<Main isWideLayout={ true }>
```
My apologies, but the prop is called `wideLayout`, not `isWideLayout` as I mistakenly said before. Corrected in d34c50a.

lamosty commented on client/signup/jetpack-connect/plans.jsx:
```diff
@@ -156,16 +147,11 @@ const Plans = React.createClass( {
 							steps={ 3 } />
 
 						<div id="plans" className="plans has-sidebar">
```
This gives me eslint warning. Maybe we could name the `div` above: `jetpack-connect__group` or `jetpack-connect__container` so we can call this `jetpack_connect__plans`. That should silence the warnings. :)

johnHackworth replied:
yeah, all those css classes were added before the new lint check for their formatting... I think we should address those warnings in a separate PR to not make this more complex and (potentially) style modifying

lamosty commented:
I think this is ready to go. :shipit:
